File: app/graphql/queries/payments.rb
module Queries
  module Payments
    extend ActiveSupport::Concern

    included do
      field :payments, [Types::Payment], null: false do
        description 'Get all payments'
        argument :year, GraphQL::Types::ID, required: true
      end

      def payments(year:)
        festival = ::Festival.by_year(year).first
        ::Payment
          .includes(registration: :user)
          .references(:registration)
          .where(registrations: { festival_id: festival.id })
      end
    end
  end
end

File: app/javascript/queries/payments.js
import gql from 'graphql-tag'

export default gql`
  query Payments($year: ID!) {
    payments(year: $year) {
      id
      type
      amount
      state
      createdAt
      registration {
        id
        user {
          id
          name
        }
      }
    }
  }
`